failsafe/circuit_breaker.py

```diff
@@ -40,14 +40,14 @@ def allows_execution(self):
 
     def record_success(self):
         """
-        Stores an execution success in the state of the CircuitBreaker
+        Function called when the execution succeeded
```

Comment:
Yes, I agree that we can remove docs from the "*State" classes. The thing is that `ClosedState` and `OpenState` classes are internal and user of the library should not use them directly and thus read their docstrings. The implementation of those classes is simple, so we do not need docs for contributors.. I think. :)

However, I would be fine also with your suggested solution, so feel free to choose whatever you prefer.

Also, I think we should name those classes `_ClosedState` and `_OpenState` to make it obvious that they are for internal use only.. as suggested by PEP8. You can do it in this MR if you want to, but we can also do it sometime later...